editor/filesystem_dock.cpp

```diff
@@ -599,6 +599,52 @@ void FileSystemDock::_update_files(bool p_keep_selection) {
 	}
 }
 
+String _extract_trailing_number(String name) {
+	String nums;
+	for (int i = name.length() - 1; i >= 0; i--) {
```

Comment:
Just being difficult here... But it might be nicer to write this as something like this (untested):

```
    int i = name.length()-1;                                                                                                                                                
    while(i >= 0 && name[i] >= '0' && name[i] <= '9') {
        num = String::chr(name[i]) + num;
         i--;
    }
```

Reply:
I extracted this code from https://github.com/godotengine/godot/blob/b6b722d5c1cb71ed6370a0673c52684e1d7ce724/scene/main/node.cpp#L1023-L1032 but I'm ok to change it.
I'll test your update and commit it.